code/Workable.php

```diff
@@ -164,6 +164,23 @@ public function callHttpClient(string $url, array $params = [], string $method =
 
             throw $e;
         }
+        
+       // remaining retries within the current interval.
+       $remainingRequests = (int) $response->getHeader('X-Rate-Limit-Remaining')[0];
+       $sleepInterval = 10;
+
+        // If we are about to hit the rate-limit, we wait for the next limit interval, or wait 10 seconds.
+        if ($remainingRequests === 1) {
```

Comment:
If this value is `1`, that means we got a successful response, if I'm understanding this correctly - we have 1 request available to us in the time period. The idea is that this then sleeps to reset the count, the executes the query again?

Would simply returning the response body be an option here?

Reply:
I've changed my approach. Rate limit is caught in the catch block. Not outside of the try/catch. Avoids having to do a check in the catch and outside of it.